daemons/controld/controld_messages.c

```diff
@@ -831,9 +845,15 @@ handle_ping(const xmlNode *msg)
     // @TODO maybe do some checks to determine meaningful status
     crm_xml_add(ping, PCMK_XA_RESULT, "ok");
```

Comment:
I can tell how the existing logic directly considers PCMK__XA_SRC from a ping reply to be the name of the DC, but technically would it make more sense to instead reflect the situation with a different value of PCMK_XA_RESULT? Then only if  PCMK_XA_RESULT is "ok", crmadmin considers PCMK__XA_SRC to be the DC?

Reply:
That would make sense, but compatibility would be a problem with mixed versions (either a rolling upgrade or remote nodes). Also, the ping op is used for other queries besides checking the DC, so those uses would have to handle the new result value as well.